### PR TITLE
GPT answers: bugs and UI fixes & production launch

### PR DESCRIPTION
**Pull request by yoavf**
This launches GPT in Help Center (instead of Sibyl) via the feature flag.

## Proposed Changes

* Enable in help-center
* Also enable in atomic (via proxy endpoint)
* Bring in some of the changes from #76935, including UI changes and code improvements. See [the commit list](https://github.com/Automattic/wp-calypso/pull/77076/commits) for details. The most notable difference is that we no longer display the GPT generated URLs, and adding a blurb to set expectations about the GPT answer while the answer loads.

### Screenshots

Screenshot 2023-05-21 at 11 38 59

Screenshot 2023-05-21 at 12 33 12

## Testing Instructions

* Load Helpcenter and click "Still Need Help"

Testing all the flows:
 - [x] With feature flag disabled (just in case, to make sure we haven't broke anything in Sibyl)
 - [x] On Atomic
 - [x] Different possible user behaviors in the GPT screen: 
   - Before the GPT answer loads: 
      -  [x] clicking back 
      -  [x] clicking cancel.
   - After the GPT answer loads
     -  [x] clicking back 
     -  [x] Clicking close
     -  [x] Clicking send ("still chat with/email us");
     -  [x] Thumbs Up
     -  [x] Thumbs Down

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by alshakero — APPROVED** [review]
Neat! The code looks great. Just looping @gravityrail in because he just [finished something](https://github.com/Automattic/wp-calypso/pull/76734) that maybe slightly orthogonal with this.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7998035

Thank you @yoavf for including a screenshot in the description! This is really helpful for our translators.

File: packages/help-center/src/components/help-center-contact-form.tsx [hunk 1]
```diff
@@ -310,7 +311,10 @@ export const HelpCenterContactForm = () => {
 			section: sectionName,
 		} );
 
+		const savedCurrentSite = currentSite;
 		resetStore();
+		setSite( savedCurrentSite );
+
 		navigate( '/' );
```
Comment (klimeryk) [inline packages/help-center/src/components/help-center-contact-form.tsx:318]: Maybe we should instead just call `setShowHelpCenter( false )` to close the Help Center since the user should be done?

File: packages/help-center/src/components/help-center-contact-form.tsx [hunk 2]
```diff
@@ -310,7 +311,10 @@ export const HelpCenterContactForm = () => {
 			section: sectionName,
 		} );
 
+		const savedCurrentSite = currentSite;
```
Comment (klimeryk) [inline packages/help-center/src/components/help-center-contact-form.tsx:314]: I'm curious - why is this needed? `resetStore` should put the store in correct, clear state. I'm worried of possible side-effects if we modify it like this.

File: apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-jetpack-search-ai.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	 */
+	public function register_rest_route() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/search',
+			array(
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_search_results' ),
+				'permission_callback' => array( $this, 'permission_callback' ),
+			)
```
Comment (klimeryk) [inline apps/editing-toolkit/editing-toolkit-plugin/help-center/class-wp-rest-help-center-jetpack-search-ai.php:35]:
```suggestion
				'permission_callback' => array( $this, 'permission_callback' ),
				'args' => array( 'query' => ...
			)
```
Probably a good idea to add `args` declaration of the parameters this endpoint supports. Especially since at the very least `query` is required. Although, I get that it feels kinda like code duplication between this and the WPCOM backend's declaration :/

File: packages/help-center/src/components/help-center-gpt.tsx [hunk 4]
```diff
@@ -123,21 +126,11 @@ export function HelpCenterGPT() {
...
-					<p>{ line }</p>
-				) ) }
-			</div>
-			<h3 id="help-center--contextual_help" className="help-center__section-title">
-				{ __( 'AI generated response:', __i18n_text_domain__ ) }
-			</h3>
+			<h1 id="help-center--contextual_help" className="help-center__section-title">
+				{ __( 'Quick response:', __i18n_text_domain__ ) }
+			</h1>
 			{ isGPTError && (
```
Comment (klimeryk) [inline packages/help-center/src/components/help-center-gpt.tsx:134]: Screenshot 2023-05-22 at 12 37 13
If/when we get to this error state, maybe the "Still chat with us" button should be primary (instead of `Close`)? Or maybe it'd better UX (and actually easier to implement) to have an effect dependent on `isGPTError` and when it's `true`, we call a callback from the parent component, that will could be a refactored (or part of) this method: https://github.com/Automattic/wp-calypso/blob/bd4871a4dbf207b391c46be4135255ab3ba493ab/packages/help-center/src/components/help-center-contact-form.tsx#L338. So that it'd do all the necessary actions, but automatically for the user. _Alternatively_, `navigate` back to the contact form, but with a parameter that will auto-submit it.

**Review by klimeryk — APPROVED** [review]
Looks like the `Button` from `wordpress/components` has some visual quirks in the context of Help Center:

https://github.com/Automattic/wp-calypso/assets/3392497/44654315-0924-402f-8728-19b9771559b1

Might be the rules here: https://github.com/Automattic/wp-calypso/blob/bd4871a4dbf207b391c46be4135255ab3ba493ab/packages/help-center/src/components/help-center-contact-form.scss#L117

While testing, I noticed that we send a request to `https://public-api.wordpress.com/wpcom/v2/help/search/wpcom?query=...` on basically every keystroke. Digging a bit more looks like these two lines need to use `debouncedMessage` instead:

https://github.com/Automattic/wp-calypso/blob/bd4871a4dbf207b391c46be4135255ab3ba493ab/packages/help-center/src/components/help-center-contact-form.tsx#L653

https://github.com/Automattic/wp-calypso/blob/bd4871a4dbf207b391c46be4135255ab3ba493ab/packages/help-center/src/components/help-center-contact-form.tsx#L783

cc @gravityrail, looks like it might have been introduced in #76734

Super nitpick, but the dot at the end of this string is bothering me 😅 

https://github.com/Automattic/wp-calypso/blob/bd4871a4dbf207b391c46be4135255ab3ba493ab/packages/help-center/src/components/help-center-contact-form.tsx#LL581C13-L581C13

Either it should be `Gathering quick response` (to be consistent with other action buttons) or maybe `Gathering quick response…` (since it's loading state).

Tested in Atomic - worked as expected 👍 

Overall, no blockers, feel free to launch as-is and make follow-ups changes 🙇

File: packages/help-center/src/components/help-center-gpt.tsx (hunk 4 above)
Comment (yoavf) [reply packages/help-center/src/components/help-center-gpt.tsx:134]: Good idea. Seems like it's a small enough change.

File: packages/help-center/src/components/help-center-contact-form.tsx (hunk 2 above)
Comment (yoavf) [reply packages/help-center/src/components/help-center-contact-form.tsx:314]: resetStore also unsets the currentSite. It's empty without this.
Try getting a response, then hitting "close" then trying again - the form will be broken.

File: packages/help-center/src/components/help-center-contact-form.tsx (hunk 1 above)
Comment (yoavf) [reply packages/help-center/src/components/help-center-contact-form.tsx:318]: I'll put that one for a separate PR - I'm just worried of the side effects.

File: packages/help-center/src/components/help-center-contact-form.tsx (hunk 2 above)
Comment (klimeryk) [reply packages/help-center/src/components/help-center-contact-form.tsx:314]: Right, but where did the site come from in the beginning? Feels like that logic is not being triggered and that should be addressed "from the beginning/front", not at the end/cleanup phase. What if the user switches the site in the meantime - will the saved site remain the same in this context?

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7998035).
